frontend/src/components/pages/RegistrantExperience/PersonalInfo/index.tsx

```diff
@@ -1,26 +1,110 @@
 import React from "react";
-import { Box, Checkbox, Text } from "@chakra-ui/react";
+import { SmallAddIcon } from "@chakra-ui/icons";
+import { Box, Button, Divider, Text, useToast, VStack } from "@chakra-ui/react";
+import { RegistrantExperienceCamper } from "../../../../types/CamperTypes";
+import { usePersonalInfoHook } from "./personalInfoReducer";
+import CamperCard from "./CamperCard";
+import { PersonalInfoActions } from "../../../../types/PersonalInfoTypes";
+import ContactCard from "./ContactCard";
+import { CampSession } from "../../../../types/CampsTypes";
 
 type PersonalInfoProps = {
-  isChecked: boolean;
-  toggleChecked: () => void;
+  nextBtnRef: React.RefObject<HTMLButtonElement>;
+  campers: RegistrantExperienceCamper[];
+  campName: string;
+  campSessions: CampSession[];
+  setCampers: React.Dispatch<
+    React.SetStateAction<RegistrantExperienceCamper[]>
+  >;
+};
+
+const checkSpaceAvailable = (
+  campSessions: CampSession[],
+  campers: RegistrantExperienceCamper[],
+): boolean => {
+  // validate whether there is enough space for a new camper
+  let spaceAvailable = 100000000;
+  /* eslint-disable-next-line */
+  for (const campSession of campSessions) spaceAvailable = Math.min(spaceAvailable,campSession.capacity - campSession.campers.length - campers.length);
+  return spaceAvailable > 0;
 };
 
 const PersonalInfo = ({
-  isChecked,
-  toggleChecked,
+  nextBtnRef,
+  campers,
+  campName,
+  campSessions,
+  setCampers,
 }: PersonalInfoProps): React.ReactElement => {
+  const toast = useToast();
+  const setPersonalInfo = usePersonalInfoHook(setCampers);
   return (
-    <Box>
-      <Text textStyle="displayXLarge">Camp Registration</Text>
-      <Text>Personal Info</Text>
+    <Box pb={14}>
+      <Text textStyle="displayXLarge">{campName} Registration</Text>
+      <Text
+        color="#10741A"
+        py={7}
+        textStyle={{ sm: "xSmallMedium", md: "xSmallBold", lg: "displayLarge" }}
+      >
+        Camper Information
+      </Text>
+      <VStack spacing={6} pb={6}>
+        {campers.map((camper, index) => (
+          <CamperCard
+            nextBtnRef={nextBtnRef}
+            key={index}
+            camper={camper}
+            setPersonalInfo={setPersonalInfo}
+            camperId={index}
+          />
+        ))}
+      </VStack>
+
+      <Button
+        w="100%"
+        backgroundColor="primary.green.100"
+        color="#ffffff"
+        onClick={() => {
+          if (checkSpaceAvailable(campSessions, campers))
+            setPersonalInfo({
+              type: PersonalInfoActions.ADD_CAMPER,
+              campSessions,
+            });
+          else
+            toast({
+              title: "Max Capacity Reached",
+              description: "Camp is currently full",
+              status: "warning",
```

Comment:
I think we also use the subtle variant

Reply:
I asked Andy about that the intended design is. It seems that we should set the `Add Additional Camper` button to be disabled on full capacity. I'll change the code accordingly.